### PR TITLE
SRE-2497: New LICENSE file as expected by OpenSSF Scorecard

### PR DESCRIPTION
**Pull request by grom72**
New LICENSE file as expected by OpenSSF Scorecard
All DAOS projects to be released under the **BSD-2-Clause Plus Patent License**

### CODE REVIEW

**Review by brianjmurrell — COMMENTED** [review]
Looks good, ~~but should you also add the other files (`SECURITY.md` at least) that IPAS is requiring by the end of this quarter?~~ NVM, I see that IPAS added https://github.com/daos-stack/.github/blob/master/SECURITY.md which I guess covers all of our projects at once.

**Review by brianjmurrell — DISMISSED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -38,6 +38,6 @@
  */
 // To use a test branch (i.e. PR) until it lands to master
 // I.e. for testing library changes
-//@Library(value="pipeline-lib@your_branch") _
+@Library(value="pipeline-lib@grom72/SRE-2453") _
```
Comment (ryon-jensen) [inline Jenkinsfile]: this will of course need to change before landing.

**Review by ryon-jensen — DISMISSED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (grom72) [reply Jenkinsfile]: Done

**Review by ryon-jensen — DISMISSED** [review]

File: pmdk.spec
Comment (brianjmurrell) [inline pmdk.spec:1]: Shouldn't https://github.com/daos-stack/pmdk/blob/cc5cb682e8255a877735523220ee573fb3395dba/pmdk.spec#L47 be **BSD-2-Clause-Patent** or **BSD-3-Clause**?  But TBH, given the overall concerns with having different licenses throughout this file, I'm thoroughly confused which it should be.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
Why are the `Jenkinsfile` and the `LICENCE` file referring to **BSD-2-Clause-Patent**/**BSD-2-Clause Plus Patent License** but the other changed files here refer to **BSD-3-Clause**?

File: pmdk.spec
Comment (grom72) [reply pmdk.spec:1]: My mistake. It should be BSD-3-Clause

File: pmdk.rpmlintrc [hunk 2]
```diff
@@ -1,3 +1,6 @@
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+# Copyright (c) 2019-2024 Intel Corporation
```
Comment (janekmi) [inline pmdk.rpmlintrc]:
```suggestion
# Copyright (c) 2022-2024 Intel Corporation
```

**Review by janekmi — CHANGES_REQUESTED** [review]

File: pmdk.spec [hunk 3]
```diff
@@ -1,3 +1,5 @@
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+# Copyright (c) 2019-2024 Intel Corporation
```
Comment (janekmi) [inline pmdk.spec]: Assuming this file started as a copy-paste from the PMDK repo I believe the correct copyright range is as follows:

```suggestion
# Copyright (c) 2017-2024 Intel Corporation
```

**Review by janekmi — CHANGES_REQUESTED** [review]

File: pmdk.spec (hunk 3 above)
Comment (grom72) [reply pmdk.spec]: Done

File: pmdk.rpmlintrc (hunk 2 above)
Comment (grom72) [reply pmdk.rpmlintrc]: Done

**Review by janekmi — APPROVED** [review]

**Review by osalyk — APPROVED** [review]

File: pmdk.spec [hunk 4]
```diff
@@ -42,7 +44,7 @@ Version:    %{major}.%{minor}.%{bugrelease}%{?prerelease:~%{prerelease}}
 Release:    %{buildrelease}%{?dist}
 Summary:    Persistent Memory Development Kit
 Group:      System Environment/Libraries
-License:    BSD
+License:    BSD-3-Clause
```
Comment (brianjmurrell) [inline pmdk.spec:47]: This seems to contradict what you [said previously](https://github.com/daos-stack/pmdk/pull/40/files#r1776539422).

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

File: pmdk.spec (hunk 4 above)
Comment (grom72) [reply pmdk.spec:47]: Fix the comment above. PMDK is distributed with the BSD-3-Clause license.
DAOS's spec that creates PMDK rpm is distributed under the BSD-2-Clause-Patent license.

Comment (janekmi) [reply pmdk.spec:47]: I would say, it is correct. @brianjmurrell do you have any further doubts?

Comment (brianjmurrell) [reply pmdk.spec:47]: Seems fine.  Confusion between PMDK and the RPM packaging of it.

**Review by brianjmurrell — APPROVED** [review]

**Pull request merged**